lib/mobility/plugins/attributes.rb

```diff
@@ -44,24 +48,17 @@ def mobility_attribute?(name)
           mobility_attributes.include?(name.to_s)
         end
 
-        # Register a new attribute name. Public, but treat as internal.
-        # @param [String, Symbol] Attribute name
-        def register_mobility_attribute(name)
-          (self.mobility_attributes << name.to_s).uniq!
-        end
-
-        def inherited(klass)
-          super
-          mobility_attributes.each { |name| klass.register_mobility_attribute(name) }
-        end
-
-        protected
-
         # Return translated attribute names on this model.
         # @return [Array<String>] Attribute names
         def mobility_attributes
           @mobility_attributes ||= []
         end
+
+        def inherited(klass)
+          super
+          attrs = mobility_attributes
+          klass.class_eval { @mobility_attributes = attrs }
```

Comment:
This is not right, we should be duping `attrs` otherwise both classes are pointing at the same object.